src/galois/_polys/_irreducible.py

```diff
@@ -180,7 +264,99 @@ def _deterministic_search(field, start, stop, step) -> Poly | None:
     return None
 
 
-def _random_search(order, degree) -> Poly:
+@functools.lru_cache(maxsize=4096)
+def _minimum_terms(order: int, degree: int, test: str) -> int:
```

Comment:
This function is used both by `irreducible_poly()` and `primitive_poly()`. Do you want to keep it in `_irreducible.py` or maybe move it to `_poly.py`?

Reply:
Same comment for `_deterministic_search_fixed_terms()`

Reply:
Yes, good point. I will move those common functions into a more common place, like `_poly.py`.